### ISSUE TITLE
cifar10_cnn.py went wrong with tensorflow backend

### ISSUE THREAD

**Issue by hushell** [fchollet/keras#3769]
I am using c58bcc2c02ff40e602a05e3eef243757d0f34acf with tensorflow-0.10 and theano-0.9. 

[cifar10_cnn.py](https://github.com/fchollet/keras/blob/master/examples/cifar10_cnn.py) works well with theano but gets weird for tensorflow. The problem is the loss doesn't change much even after 80 epochs. 

> Train on 50000 samples, validate on 10000 samples
> Epoch 1/200
> 50000/50000 [==============================] - 16s - loss: 2.3039 - acc: 0.0983 - val_loss: 2.3034 - val_acc: 0.1000
> Epoch 2/200
> 50000/50000 [==============================] - 16s - loss: 2.3036 - acc: 0.0987 - val_loss: 2.3034 - val_acc: 0.1000
> Epoch 3/200
> 50000/50000 [==============================] - 16s - loss: 2.3034 - acc: 0.0993 - val_loss: 2.3028 - val_acc: 0.1000
> Epoch 4/200
> 50000/50000 [==============================] - 16s - loss: 2.3034 - acc: 0.0992 - val_loss: 2.3030 - val_acc: 0.1000
> Epoch 5/200
> 50000/50000 [==============================] - 16s - loss: 2.3033 - acc: 0.0983 - val_loss: 2.3031 - val_acc: 0.1000
> Epoch 6/200
> 50000/50000 [==============================] - 16s - loss: 2.3033 - acc: 0.0978 - val_loss: 2.3031 - val_acc: 0.1000
> Epoch 7/200
> 50000/50000 [==============================] - 16s - loss: 2.3034 - acc: 0.1006 - val_loss: 2.3033 - val_acc: 0.1000
> Epoch 8/200
> 50000/50000 [==============================] - 16s - loss: 2.3033 - acc: 0.1020 - val_loss: 2.3030 - val_acc: 0.1000
> Epoch 9/200
> 50000/50000 [==============================] - 16s - loss: 2.3034 - acc: 0.1004 - val_loss: 2.3031 - val_acc: 0.1000
> Epoch 10/200

**Comment by fchollet** [comment]
Try a native TensorFlow convnet. Verify that it doesn't work either.

**Comment by deependersingla** [comment]
@fchollet  Any update? I had the same issue.

**Comment by felixlaumon** [comment]
I encountered this issue today too. `examples/cifar10_cnn.py` stays at loss of 2.3 and acc of ~0.1. Then I tried running both `examples/mnist_cnn.py` and `python -m tensorflow.models.image.mnist.convolutional` which both converges to ~97% acc. After that `examples/cifar10_cnn.py` somehow works correctly which the model will converge. Very strange.

**Comment by felixlaumon** [comment]
Actually this seems to be a weight initialization issue (!?) Both `he_uniform` and `he_normal` is able to escape the plateau 5 out of 5 times for me in 10% into the first epoch.
